src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -328,13 +333,29 @@ public function url($path)
      *
      * @param  \League\Flysystem\AwsS3v3\AwsS3Adapter  $adapter
      * @param  string  $path
+     * @param  int|string|\DateTime $expires The time at which the URL should
+     *                                      expire. This can be a Unix
+     *                                      timestamp, a PHP DateTime object,
+     *                                      or a string that can be evaluated
+     *                                      by strtotime().
      * @return string
      */
-    protected function getAwsUrl($adapter, $path)
+    protected function getAwsUrl($adapter, $path, $expires)
```

Comment:
WTH, I don't know how I forgot this, I pulled this feature once to 5.4 and it had no errors, I was in a hurry while pushing this.